lib/rules/block-attributes-order.js

```diff
@@ -0,0 +1,320 @@
+/**
+ * @fileoverview enforce ordering of block attributes
+ * @author Wenlu Wang
+ */
+'use strict'
+const utils = require('../utils')
+const { parseSelector } = require('../utils/selector')
+
+/**
+ * @enum {string}
+ */
+const WELL_KNOWN_TEMPLATE_ATTRS = {
+  functional: 'functional',
+  lang: 'lang',
+  src: 'src'
+}
+
+/**
+ * @enum {string}
+ */
+const WELL_KNOWN_SCRIPT_ATTRS = {
+  lang: 'lang',
+  setup: 'setup',
+  src: 'src'
+}
+
+/**
+ * @enum {string}
+ */
+const WELL_KNOWN_STYLE_ATTRS = {
+  scoped: 'scoped',
+  module: 'module',
+  lang: 'lang',
+  src: 'src'
+}
+
+/**
+ * @typedef UserOrderOption
+ * @property { string } element
+ * @property { (string | string[])[] } order
+ */
+
+/**
+ * @typedef {import('../utils/selector').VElementSelector} VElementSelector
+ */
+
+/**
+ * @typedef OrderOption
+ * @property {string} element
+ * @property {(string | string[])[]} order
+ * @property {(RegExp | RegExp[])[]} regexOrder
+ * @property {VElementSelector} selector
+ */
+
+/**
+ * @type {readonly UserOrderOption[]}
+ */
+const defaultOptions = Object.freeze([
+  {
+    element: 'style',
+    order: [
+      WELL_KNOWN_STYLE_ATTRS.module,
+      WELL_KNOWN_STYLE_ATTRS.scoped,
+      WELL_KNOWN_STYLE_ATTRS.lang,
+      WELL_KNOWN_STYLE_ATTRS.src
+    ]
+  },
+  {
+    element: 'script',
+    order: [
+      WELL_KNOWN_SCRIPT_ATTRS.setup,
+      WELL_KNOWN_SCRIPT_ATTRS.lang,
+      WELL_KNOWN_SCRIPT_ATTRS.src
+    ]
+  },
+  {
+    element: 'template',
+    order: [
+      WELL_KNOWN_TEMPLATE_ATTRS.functional,
+      WELL_KNOWN_TEMPLATE_ATTRS.lang,
+      WELL_KNOWN_TEMPLATE_ATTRS.src
+    ]
+  }
+])
+
+/**
+ *
+ * @param {string | string[]} item
+ * @return {RegExp | RegExp[]}
+ */
+function ordersToRegexOrders(item) {
+  return Array.isArray(item) ? item.map((x) => new RegExp(x)) : new RegExp(item)
+}
```

Comment:
Could you change it to accept more complex options by doing the following?

```suggestion
const { toRegExp } = require('../utils/regexp')

/**
 * @param {string | string[]} item
 * @return {(str: string) => boolean}
 */
function ordersToMatcher(item) {
    return Array.isArray(item) ? compileMatcher(item) : compileMatcher([item])
}

/**
 * Compile matcher
 * @param {string[]} pattern
 * @return {(str: string) => boolean}
 */
function compileMatcher(pattern) {
  const rules = []
  for (const p of pattern) {
    let negative, patternStr
    if (p.startsWith("!")) {
      // If there is `!` at the beginning, it will be parsed with a negative pattern.
      negative = true
      patternStr = p.substring(1)
    } else {
      negative = false
      patternStr = p
    }
    const regex = toRegExp(patternStr)
    rules.push({ negative, match: (str) => regex.test(str) })
  }
  return (str) => {
    // If the first rule is a negative pattern, they are considered to match if they do not match that pattern.
    let result = Boolean(rules[0]?.negative)
    for (const { negative, match } of rules) {
      if (result === !negative) {
        // Even if it matches, the result does not change, so skip it.
        continue
      }
      if (match(str)) {
        result = !negative
      }
    }
    return result
  }
}
```

This change allows users to use pattern formats such as:

- `"foo"` … Matches only the `foo` attribute name.
- `"/foo/"` … Matches attribute names that match the `/foo/` regex. That is, it matches the attribute name including `foo`.
- `"!foo"` … Exclude `foo` attribute from the matched attribute names. When used first in the array or alone, matches other than the `foo` attribute name.
- `"!/foo/"` … Excludes attributes that match the `/foo/` regex from the matched attribute names. When used first in the array or alone, matches an attribute name that does not match the `/foo/` regex.
- `["foo", "/^bar/u"]` … Matches the `foo` attribute or the attribute name that matches the `/^bar/u` regex.
- `["/^foo/u", "!foo-bar", "/^baz/u"]` … Matches an attribute name that matches `/^foo/u` and other than `foo-bar`, or an attribute name that matches `/^baz/u`.

I think the rule of other plugin will also be helpful.
https://ota-meshi.github.io/eslint-plugin-svelte/rules/sort-attributes/